api/src/index.js

```diff
@@ -23,9 +25,68 @@ apiRouter.get("/", async (req, res) => {
 
 // This nested router example can also be replaced with your own sub-router
 apiRouter.use("/nested", nestedRouter);
+app.use('/api/meals', mealRouter);
+app.use('/api/reservation', reservationRouter);
+
 
 app.use("/api", apiRouter);
 
 app.listen(process.env.PORT, () => {
   console.log(`API listening on port ${process.env.PORT}`);
 });
+//Future-meal
+app.get("/future-meals", async (req, res) => {
+  try {
+    const meals = await knex("meal").where("when", ">", knex.fn.now());
+    res.json(meals);
+  } catch (error) {
+    res.status(500).json({ error: error.message });
+  }
+});
+
+//past-meal
+app.get("/past-meals", async (req, res) => {
+  try {
+    const now = new Date();
+    const meals = await knex("meal").where("when", "<", now);
+    res.json(meals);
+  } catch (error) {
+    res.status(500).json({ error: error.message });
+  }
+});
+
+//all-meal
+app.get("/all-meals", async (req, res) => {
+  try {
+    const meals = await knex("meal").orderBy("id");
+    res.json(meals);
+  } catch (error) {
+    res.status(500).json({ error: error.message });
+  }
+});
+
+//First-meal
+app.get("/first-meal", async (req, res) => {
+  try {
+    const meal = await knex("meal").orderBy("id").first();
+    if (!meal) {
+      return res.status(404).json({ message: "No meals available" });
+    }
+    res.json(meal);
+  } catch (error) {
+    res.status(500).json({ error: error.message });
```

Comment:
which line of code would trigger this catch statement? Is it all the code or just specific lines :) Try to limit the try catch to only the statements you cant handle.